src/layers/group.rs

```diff
@@ -22,6 +20,9 @@ impl GroupLayerData {
         infinite: bool,
         map_path: &Path,
         tilesets: &[MapTilesetGid],
+        for_tileset: Option<Arc<Tileset>>,
+        reader: &mut impl ResourceReader,
+        cache: &mut impl ResourceCache,
```

Comment:
I see that in many places we're passing around both a reader and a cache. Would it make sense to pass around a loader instead?

Reply:
How does that pose a problem for passing `loader: &mut Loader` instead of the above two parameters?